File: CS/WindowsApplication3/MVVM/ViewModels/BaseViewModel.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.ComponentModel;
using System.Linq;
using DevExpress.Mvvm.POCO;

namespace MVVMSample {
    public class BaseViewModel<T>
            where T : class, new() {
        public BaseViewModel() {
            SupportNewEntities = true;
        }
        public virtual BindingList<T> Entities { get; set; }
        public virtual bool SupportNewEntities { get; set; }
        public void Add() {
            Entities.Add(new T());
        }
        public bool CanAdd() {
            return SupportNewEntities;
        }
        protected void OnSupportNewEntitiesChanged() {
            this.RaiseCanExecuteChanged(x => x.Add());
        }
        public void Clear() {
            Entities.Clear();
        }
        public void RemoveLast() {
            if (Entities.Count > 0) {
                Entities.RemoveAt(Entities.Count - 1);
            }
        }
    }
}

File: VB/WindowsApplication3/MVVM/ViewModels/BaseViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Collections.ObjectModel
Imports System.ComponentModel
Imports System.Linq
Imports DevExpress.Mvvm.POCO

Namespace MVVMSample
    Public Class BaseViewModel(Of T As {Class, New})
        Public Sub New()
            SupportNewEntities = True
        End Sub
        Public Overridable Property Entities() As BindingList(Of T)
        Public Overridable Property SupportNewEntities() As Boolean
        Public Sub Add()
            Entities.Add(New T())
        End Sub
        Public Function CanAdd() As Boolean
            Return SupportNewEntities
        End Function
        Protected Sub OnSupportNewEntitiesChanged()
            Me.RaiseCanExecuteChanged(Sub(x) x.Add())
        End Sub
        Public Sub Clear()
            Entities.Clear()
        End Sub
        Public Sub RemoveLast()
            If Entities.Count > 0 Then
                Entities.RemoveAt(Entities.Count - 1)
            End If
        End Sub
    End Class
End Namespace
